gr-utils/python/modtool/templates/templates.py

```diff
@@ -563,26 +563,25 @@ def test_001_t(self):
   label: ...
   dtype: ...
 
-#  Make one 'inputs' list entry per input. Sub-entries of dictionary:
+#  Make one 'inputs' and one 'outputs' list entry per input/output. Keys include:
 #      * label (an identifier for the GUI)
-#      * domain
-#      * dtype
-#      * vlen
-#      * optional (set to 1 for optional inputs) 
+#      * domain (optional - stream or message. Default is stream)
+#      * dtype (e.g. int, float, complex, byte, short, xxx_vector, ...)
+#      * vlen (optional - data stream vector length. Default is 1)
+#      * optional (optional - set to 1 for optional inputs. Default is 0)
 inputs:
 - label: ...
   domain: ...
   dtype: ...
   vlen: ...
+  optional: ...
 
-#  Make one 'outputs' list entry per output. Sub-entries of dictionary:
-#      * label (an identifier for the GUI)
-#      * dtype
-#      * vlen
-#      * optional (set to 1 for optional inputs) 
+outputs:
 - label: ...
   domain: ...
-  dtype: ... #!-- e.g. int, float, complex, byte, short, xxx_vector, ...--
+  dtype: ...
+  vlen: ...
+  optional: ...
 
 file_format: 1
```

Comment:
A comment on file_format would be nice here.  It is a required field but there is no documentation anywhere as to why it is needed or what it means.  Perhaps just a "required field" comment would be helpful so people don't figure out the hard way.

Reply:
The file format specifies the syntax version which makes handling future changes easier. With the XML files this always involved some guess working

Reply:
I agree. Comment added.